### PR TITLE
[Merged by Bors] - ET-4004 Return bad request if document does not exists

### PR DESCRIPTION
**Pull request by acrrd**
Return an error if some of the documents do not exist.
The documentation was mentioning that the endpoint does not return an error and then contradicts itself by stating that it could return 400. The part around the idempotency of the method was also wrong. Being idempotent does not mean that a method does not return an error if something does not exist.
The HTTP standard requires DELETE methods to be idempotent and it specify that it is in respect to the state of the server not the response. 
```
A request method is considered "idempotent" if the intended EFFECT ON THE SERVER of multiple identical requests with that method is the same as the effect for a single such request. Of the request methods defined by this specification, PUT, DELETE, and safe request methods are idempotent.
```

Returning an error makes `DELETE /documents/<id>` behaving like `GET /documents/<id>`  where we return 400 if the document does not exists.

### CODE REVIEW

File: web-api/src/storage.rs [hunk 1]
```diff
@@ -71,6 +71,12 @@ pub(crate) enum DeletionError {
     PartialFailure { errors: Vec<DocumentIdAsObject> },
 }
 
+impl From<sqlx::Error> for DeletionError {
+    fn from(e: sqlx::Error) -> Self {
+        DeletionError::General(e.into())
+    }
+}
```
Comment (janpetschexain) [inline web-api/src/storage.rs:78]: :art: we don't have to impl this conversion manually, instead we can annotate the variants of the `DeletionError` like
```rust
#[derive(Debug, From)]
pub(crate) enum DeletionError {
    #[from(types(sqlx::Error))]
    General(Error),
    #[from]
    PartialFailure { errors: Vec<DocumentIdAsObject> },
}
```

**Review by janpetschexain — APPROVED** [review]

File: web-api/src/storage.rs [hunk 2]
```diff
@@ -71,6 +71,12 @@ pub(crate) enum DeletionError {
     PartialFailure { errors: Vec<DocumentIdAsObject> },
 }
 
+impl From<sqlx::Error> for DeletionError {
```
[the remark this answers is not in the document]
Comment (acrrd) [reply web-api/src/storage.rs:74]: Is not about moving the error to another place, the storage is still using it, and it depends on application errors that the storage should not know about.

File: web-api/src/storage.rs (hunk 1 above)
Comment (acrrd) [reply web-api/src/storage.rs]: Nice, thanks

**Comment by acrrd** [comment]
bors merge

**Pull request closed without merging**